backend/infrahub/graphql/mutations/repository.py

```diff
@@ -181,13 +182,14 @@ async def mutate_update(
 
 def cleanup_payload(data: InputObjectType | dict[str, Any]) -> None:
     """If the input payload contains an http URL that doesn't end in .git it will be added to the payload"""
-    http_without_dotgit = r"^(https?://)(?!.*\.git$).*"
-    if (
-        data.get("location")
-        and data["location"].get("value")
-        and re.match(http_without_dotgit, data["location"]["value"])
-    ):
-        data["location"]["value"] = f'{data["location"]["value"]}.git'
+    if config.SETTINGS.git.append_git_suffix:
+        http_without_dotgit = r"^(https?://)(?!.*\.git$).*"
```

Comment:
is git supported over https only transport?

Reply:
This is just a regex that would look at the repository URL and optionally add a suffix if the URL is http or https.